nucleus/common/mbeanserver/src/main/java/org/glassfish/admin/mbeanserver/ssl/SSLClientConfigurator.java

```diff
@@ -595,15 +602,15 @@ private CipherInfo(final String cipherName,
         }
 
         public static void updateCiphers(final SSLContext sslContext) {
-            SSLServerSocketFactory factory = sslContext.getServerSocketFactory();
-            String[] supportedCiphers = factory.getDefaultCipherSuites();
-            for (int i = 0, len = supportedCiphers.length; i < len; i++) {
-                String s = supportedCiphers[i];
-                ciphers.put(s, new CipherInfo(s, (short) (SSL3 | TLS)));
-            }
+            ciphers.putAll(getInstance().getSupportedCipherSuites());
         }
+
         public static CipherInfo getCipherInfo(final String configName) {
-            return ciphers.get(configName);
+            CipherInfo ci = ciphers.get(configName);
+            if (ci == null) {
+                ciphers.putAll(getInstance().getSupportedCipherSuites());
```

Comment:
Shouldn't this retry to get the cipher by configName? Looks a bit odd to fill the supported cipher suites but not try to return a cipher if it happened to be added.

Reply:
This method tries to get a cipher by its name. Since this name comes from the admin console GUI as a result of the user's selection in the associated list box, this name can't be unknown. Hence, if the the name passed as the input argument isn't find in the ciphers map, then we assume that the cipher map isn't initialized with the Java standard available default cipher suites and, consequently, we initialize it here.